src/cpp/rtps/common/Time_t.cpp

```diff
@@ -19,6 +19,12 @@
 
 using namespace eprosima::fastrtps;
 
+constexpr uint64_t C_FRACTIONS = 4294967296;
+constexpr uint64_t C_SECONDS = 1000000000;
+
+#define frac_to_nano(x) (static_cast<uint32_t>(((x) * C_SECONDS) / C_FRACTIONS))
```

Comment:
Function-like macros should have comments regarding expected input. For instance, does this one expects x to be uint64_t, or may it receive uint32_t or a signed value or a floating point?

Reply:
As commented, I'll replace the macros with "compilation-unit" inline functions.